src/Mantasflowers.WebApi/Controllers/UserController.cs

```diff
@@ -0,0 +1,226 @@
+﻿using FirebaseAdmin.Auth;
+using Mantasflowers.Persistence.Authentication;
+using Mantasflowers.WebApi.Extensions;
+using Mantasflowers.WebApi.Responses;
+using Mantasflowers.WebApi.Setup.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Mantasflowers.WebApi.Controllers
+{
+    [ApiController]
+    [Route("/firebase")]
+    public class UserController : ControllerBase
+    {
+        private readonly ILogger _logger = Log.ForContext<UserController>();
+
+        private readonly FirebaseContext _fbContext;
+
+        private readonly IHttpClientFactory _clientFactory;
+
+        private readonly string _requestUri;
+
+        public UserController(FirebaseContext fbContext, IHttpClientFactory clientFactory, IOptionsMonitor<WebApiKey> optionsAccessor)
+        {
+            _fbContext = fbContext;
+            _clientFactory = clientFactory;
+            _requestUri = $"https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key={optionsAccessor.CurrentValue.Value}";
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="email"></param>
+        /// <param name="password"></param>
+        /// <returns></returns>
+        [HttpGet("get-token")]
+        public async Task<IActionResult> GetTokenAsync(string email, string password)
```

Comment:
I really don't like the facts that the entire business logic is in a controller. Would suggest you to create a `FirebaseService` and 90% of this stuff there. Also injecting httpClients should happen there

Reply:
Some context: since initially firebase was defined in Mantasflowers.Persistence assembly, there was some kind of package conflict, which prevented from using System.Net.Http for HttpClients. My guess, would be EF.